qiime2/metadata/metadata.py

```diff
@@ -100,6 +116,7 @@ def __init__(self, index):
         self._ids = tuple(index)
 
         self._artifacts = []
+        self._non_artifact_metadata = []
```

Comment:
I think we will need to update the eq/neq dunder methods to include these, right?